packages/client/components/AdhocTeamMultiSelect/AdhocTeamMultiSelect.tsx

```diff
@@ -0,0 +1,157 @@
+import graphql from 'babel-plugin-relay/macro'
+import React from 'react'
+import {useFragment} from 'react-relay'
+import Autocomplete from '@mui/material/Autocomplete'
+import TextField from '@mui/material/TextField'
+
+import {AdhocTeamMultiSelect_viewer$key} from '../../__generated__/AdhocTeamMultiSelect_viewer.graphql'
+import {Send as SendIcon} from '@mui/icons-material'
+import {Chip} from '../../ui/Chip/Chip'
+
+export type Option = {
+  id: string | null
+  label: string
+  email: string
+  picture: string | null
+  organizationIds: string[]
+}
+
+interface Props {
+  viewerRef: AdhocTeamMultiSelect_viewer$key
+  value: Option[]
+  onChange: (values: Option[]) => void
+}
+
+const autocompleteEmail = (input: string, domain: string) => {
+  // If the input already ends with the provided domain, or is just "@" then return it as-is.
+  if (input.endsWith(`@${domain}`) || input === '@') {
+    return input
+  }
+
+  // Split the input by "@" to separate the username and domain parts.
+  const [username, potentialDomain] = input.split('@')
+
+  // If there's a domain part and it's not empty, return the input as-is.
+  if (potentialDomain && potentialDomain.length > 0) {
+    return input
+  }
+
+  // If the input doesn't have a domain part, append the provided domain.
+  return `${username}@${domain}`
+}
+
+export const AdhocTeamMultiSelect = (props: Props) => {
+  const {viewerRef, onChange, value} = props
+  const viewer = useFragment(
+    graphql`
+      fragment AdhocTeamMultiSelect_viewer on User {
+        id
+        email
+        organizations {
+          id
+          organizationUsers {
+            edges {
+              node {
+                user {
+                  id
+                  preferredName
+                  picture
+                  email
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    viewerRef
+  )
+
+  const {email, organizations} = viewer
+  const viewerDomain = email.slice(email.indexOf('@') + 1)
+
+  const usersMap: {[key: string]: Option} = {}
+
+  organizations.forEach((org) => {
+    org.organizationUsers.edges.forEach((edge) => {
+      const user = edge.node.user
+      if (!usersMap[user.id]) {
+        usersMap[user.id] = {
+          id: user.id,
+          label: user.preferredName,
+          email: user.email,
+          picture: user.picture,
+          organizationIds: [org.id]
+        }
+      } else {
+        usersMap[user.id]?.organizationIds.push(org.id)
+      }
+    })
+  })
+
+  const options = Object.values(usersMap)
+
+  const createCustomOption = (option: string): Option => {
+    return {
+      label: `Invite ${option}`,
+      email: option,
+      organizationIds: [],
+      picture: null,
+      id: null
+    }
+  }
+
+  return (
+    <Autocomplete
+      multiple
```

Comment:
-1 While this would be a nice input field in other parts of the app and works well, we should only allow adding 1 user for 1:1 meeting type

Reply:
@Dschoordsch I was thinking to first finish this as a generic component with full functionality, then in some next PR add a validation and one option restriction, If this is OK